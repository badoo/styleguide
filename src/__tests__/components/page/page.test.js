import React from 'react';
import { create, act } from 'react-test-renderer';
import 'jest-styled-components';
import Page from '../../../components/page/page';

describe('Page tests:', () => {
    const mobileUserAgent =
        'Mozilla/5.0 (iPhone; CPU iPhone OS 13_2_3 like Mac OS X) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/13.0.3 Mobile/15E148 Safari/604.1';

    it('Page render', () => {
        let root;

        act(() => {
            root = create(<Page />);
        });

        expect(root.toJSON()).toMatchSnapshot();
    });

    it('Page sidebar visibility toggled', () => {
        let root;

        act(() => {
            root = create(<Page />);
        });

        expect(root.toJSON()).toMatchSnapshot();
        expect(root.getInstance().state.isSidebarVisible).toBe(true);

        act(() => {
            root.update(<Page />);
            let instance = root.getInstance();
            instance.setSidebarVisible(false);
        });

        expect(root.getInstance().state.isSidebarVisible).toBe(false);
    });

    it('Page handleKeyDown test', () => {
        let root;

        act(() => {
            root = create(<Page />);
        });

        act(() => {
            root.update(<Page />);
            const event = { keyCode: 83 };
            let instance = root.getInstance();
            instance.handleKeyDown(event);
        });

        act(() => {
            const event = { keyCode: 83 };
            let instance = root.getInstance();

            instance.handleKeyDown(event);
        });

        expect(root.toJSON()).toMatchSnapshot();
    });

    it('Page handleScreenResize test', () => {
        let root;

        act(() => {
            root = create(<Page />);
        });

        act(() => {
            root.update(<Page />);
            const event = {
                target: {
                    navigator: {
                        userAgent: mobileUserAgent,
                    },
                },
            };
            let instance = root.getInstance();
            instance.handleScreenResize(event);
        });

        expect(root.getInstance().state.deviceViewport).toBe(true);
    });

    it('Page sidebar should be closed when mobile detected', () => {
        let root;

        act(() => {
            root = create(<Page />);
        });

        act(() => {
            root.update(<Page />);
            const event = {
                target: {
                    navigator: {
                        userAgent: mobileUserAgent,
                    },
                },
            };
            let instance = root.getInstance();
            instance.handleScreenResize(event);
        });

        expect(root.getInstance().state.isSidebarVisible).toBe(false);
    });

    it('Page sidebar should be closed when hash changed while mobile detected', () => {
        let root;

        act(() => {
            root = create(<Page />);
        });

        act(() => {
            root.update(<Page />);
            const event = {
                target: {
                    navigator: {
                        userAgent: mobileUserAgent,
                    },
                },
            };
            let instance = root.getInstance();
            instance.handleScreenResize(event);
        });

        act(() => {
            root.update(<Page />);
            let instance = root.getInstance();
            instance.setSidebarVisible(true);
        });

        act(() => {
            root.update(<Page />);
            const event = {
                target: {
                    location: {
                        hash: '#Structure-Component',
                    },
                },
            };
            let instance = root.getInstance();
            instance.handleHashChange(event);
        });

        expect(root.getInstance().state.isSidebarVisible).toBe(false);
    });
});
