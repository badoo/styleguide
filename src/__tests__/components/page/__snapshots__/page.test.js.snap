// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page tests: Page handleKeyDown test 1`] = `
.c3 {
  -webkit-font-smoothing: antialiased;
  position: relative;
  height: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  background: #f0f0f0;
  border-right: 1px solid #f0f0f0;
  width: inherit;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin: 16px 0;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
}

.c9 {
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
  overflow-y: auto;
}

.c8 {
  width: 100%;
  padding: 0 16px;
}

.c7 {
  width: 32px;
  height: 32px;
  margin-bottom: 8px;
}

.c1 {
  position: fixed;
  z-index: 3;
  cursor: pointer;
  width: 24px;
  height: 24px;
  -webkit-transition: all 0.2s cubic-bezier(0.34,1.56,0.64,1);
  transition: all 0.2s cubic-bezier(0.34,1.56,0.64,1);
  top: 16px;
  left: 16px;
  -webkit-transform: translate3d(-50px,0px,0px);
  -ms-transform: translate3d(-50px,0px,0px);
  transform: translate3d(-50px,0px,0px);
}

.c1 svg {
  pointer-events: none;
  fill: currentColor;
}

.c2 {
  position: fixed;
  z-index: 3;
  cursor: pointer;
  width: 24px;
  height: 24px;
  -webkit-transition: all 0.2s cubic-bezier(0.34,1.56,0.64,1);
  transition: all 0.2s cubic-bezier(0.34,1.56,0.64,1);
  top: 16px;
  left: 16px;
  -webkit-transform: scale(1);
  -ms-transform: scale(1);
  transform: scale(1);
  opacity: 1;
}

.c2 svg {
  pointer-events: none;
  fill: currentColor;
}

.c5 {
  -webkit-transition: -webkit-transform 0.2s cubic-bezier(0.165,0.84,0.44,1);
  -webkit-transition: transform 0.2s cubic-bezier(0.165,0.84,0.44,1);
  transition: transform 0.2s cubic-bezier(0.165,0.84,0.44,1);
  position: fixed;
  left: 0;
  top: 0;
  bottom: 0;
  width: 300px;
  z-index: 1;
  will-change: transform;
}

.c10 {
  padding: 32px 0;
  min-width: 0;
  position: absolute;
  background: #fff;
  right: 0;
  left: 0;
  height: 100vh;
  -webkit-transition: left 0.3s cubic-bezier(0.165,0.84,0.44,1);
  transition: left 0.3s cubic-bezier(0.165,0.84,0.44,1);
  will-change: left;
}

.c0 {
  background: #fff;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c0 .c4 {
  -webkit-transform: translateX(0);
  -ms-transform: translateX(0);
  transform: translateX(0);
}

.c11 .c4 {
  -webkit-transform: translateX(-100%);
  -ms-transform: translateX(-100%);
  transform: translateX(-100%);
}

@media screen and (max-width:376px) {
  .c2 {
    right: 16px;
    left: initial;
  }
}

@media screen and (max-width:376px) {

}

@media (min-width:768px) {
  .c10 {
    padding: 0 32px;
    left: 300px;
  }
}

@media (min-width:768px) {

}

@media screen and (max-width:376px) {
  .c0 .c4 {
    right: 0;
    width: auto;
  }
}

@media screen and (max-width:376px) {
  .c11 .c4 {
    right: 0;
    width: auto;
  }
}

<div
  className="c0"
>
  <div
    className="c1"
    onClick={[Function]}
  >
    <svg
      id="open"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M2 6a1 1 0 011-1h18a1 1 0 011 1 1 1 0 01-1 1H3a1 1 0 01-1-1zm0 6a1 1 0 011-1h18a1 1 0 011 1 1 1 0 01-1 1H3a1 1 0 01-1-1zm1 5a1 1 0 00-1 1 1 1 0 001 1h18a1 1 0 001-1 1 1 0 00-1-1H3z"
      />
    </svg>
  </div>
  <div
    className="c2"
    onClick={[Function]}
  >
    <svg
      id="close"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M6.23 4.81A1 1 0 004.8 6.23L10.6 12 4.8 17.77a1 1 0 101.42 1.42L12 13.4l5.78 5.78a1 1 0 001.4-1.42L13.42 12l5.78-5.77a1 1 0 10-1.41-1.42L12 10.6 6.23 4.8z"
      />
    </svg>
  </div>
  <aside
    className="c3 c4 c5"
  >
    <div
      className="c6"
    >
      <div
        className="c7"
      >
        <svg
          viewBox="0 0 44 44"
          xmlns="http://www.w3.org/2000/svg"
        >
          <rect
            fill="#7000E3"
            height="44"
            rx="3.9"
            width="44"
          />
          <path
            d="M28 11.3c-2.3 0-4.6 1-6 3-1.4-2-3.7-3-6-3-4.3 0-7.8 3.3-7.8 7.6 0 6.5 9.3 15 13.8 15s13.8-8.5 13.8-15c0-4.4-3.5-7.7-7.8-7.7"
            fill="#FF671B"
          />
        </svg>
      </div>
      <div
        className="c8"
      />
    </div>
    <div
      className="c9"
    />
  </aside>
  <main
    className="c10"
  />
</div>
`;

exports[`Page tests: Page render 1`] = `
.c3 {
  -webkit-font-smoothing: antialiased;
  position: relative;
  height: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  background: #f0f0f0;
  border-right: 1px solid #f0f0f0;
  width: inherit;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin: 16px 0;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
}

.c9 {
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
  overflow-y: auto;
}

.c8 {
  width: 100%;
  padding: 0 16px;
}

.c7 {
  width: 32px;
  height: 32px;
  margin-bottom: 8px;
}

.c1 {
  position: fixed;
  z-index: 3;
  cursor: pointer;
  width: 24px;
  height: 24px;
  -webkit-transition: all 0.2s cubic-bezier(0.34,1.56,0.64,1);
  transition: all 0.2s cubic-bezier(0.34,1.56,0.64,1);
  top: 16px;
  left: 16px;
  -webkit-transform: translate3d(-50px,0px,0px);
  -ms-transform: translate3d(-50px,0px,0px);
  transform: translate3d(-50px,0px,0px);
}

.c1 svg {
  pointer-events: none;
  fill: currentColor;
}

.c2 {
  position: fixed;
  z-index: 3;
  cursor: pointer;
  width: 24px;
  height: 24px;
  -webkit-transition: all 0.2s cubic-bezier(0.34,1.56,0.64,1);
  transition: all 0.2s cubic-bezier(0.34,1.56,0.64,1);
  top: 16px;
  left: 16px;
  -webkit-transform: scale(1);
  -ms-transform: scale(1);
  transform: scale(1);
  opacity: 1;
}

.c2 svg {
  pointer-events: none;
  fill: currentColor;
}

.c5 {
  -webkit-transition: -webkit-transform 0.2s cubic-bezier(0.165,0.84,0.44,1);
  -webkit-transition: transform 0.2s cubic-bezier(0.165,0.84,0.44,1);
  transition: transform 0.2s cubic-bezier(0.165,0.84,0.44,1);
  position: fixed;
  left: 0;
  top: 0;
  bottom: 0;
  width: 300px;
  z-index: 1;
  will-change: transform;
}

.c10 {
  padding: 32px 0;
  min-width: 0;
  position: absolute;
  background: #fff;
  right: 0;
  left: 0;
  height: 100vh;
  -webkit-transition: left 0.3s cubic-bezier(0.165,0.84,0.44,1);
  transition: left 0.3s cubic-bezier(0.165,0.84,0.44,1);
  will-change: left;
}

.c0 {
  background: #fff;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c0 .c4 {
  -webkit-transform: translateX(0);
  -ms-transform: translateX(0);
  transform: translateX(0);
}

@media screen and (max-width:376px) {
  .c2 {
    right: 16px;
    left: initial;
  }
}

@media (min-width:768px) {
  .c10 {
    padding: 0 32px;
    left: 300px;
  }
}

@media screen and (max-width:376px) {
  .c0 .c4 {
    right: 0;
    width: auto;
  }
}

<div
  className="c0"
>
  <div
    className="c1"
    onClick={[Function]}
  >
    <svg
      id="open"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M2 6a1 1 0 011-1h18a1 1 0 011 1 1 1 0 01-1 1H3a1 1 0 01-1-1zm0 6a1 1 0 011-1h18a1 1 0 011 1 1 1 0 01-1 1H3a1 1 0 01-1-1zm1 5a1 1 0 00-1 1 1 1 0 001 1h18a1 1 0 001-1 1 1 0 00-1-1H3z"
      />
    </svg>
  </div>
  <div
    className="c2"
    onClick={[Function]}
  >
    <svg
      id="close"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M6.23 4.81A1 1 0 004.8 6.23L10.6 12 4.8 17.77a1 1 0 101.42 1.42L12 13.4l5.78 5.78a1 1 0 001.4-1.42L13.42 12l5.78-5.77a1 1 0 10-1.41-1.42L12 10.6 6.23 4.8z"
      />
    </svg>
  </div>
  <aside
    className="c3 c4 c5"
  >
    <div
      className="c6"
    >
      <div
        className="c7"
      >
        <svg
          viewBox="0 0 44 44"
          xmlns="http://www.w3.org/2000/svg"
        >
          <rect
            fill="#7000E3"
            height="44"
            rx="3.9"
            width="44"
          />
          <path
            d="M28 11.3c-2.3 0-4.6 1-6 3-1.4-2-3.7-3-6-3-4.3 0-7.8 3.3-7.8 7.6 0 6.5 9.3 15 13.8 15s13.8-8.5 13.8-15c0-4.4-3.5-7.7-7.8-7.7"
            fill="#FF671B"
          />
        </svg>
      </div>
      <div
        className="c8"
      />
    </div>
    <div
      className="c9"
    />
  </aside>
  <main
    className="c10"
  />
</div>
`;

exports[`Page tests: Page sidebar visibility toggled 1`] = `
.c3 {
  -webkit-font-smoothing: antialiased;
  position: relative;
  height: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  background: #f0f0f0;
  border-right: 1px solid #f0f0f0;
  width: inherit;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin: 16px 0;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
}

.c9 {
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
  overflow-y: auto;
}

.c8 {
  width: 100%;
  padding: 0 16px;
}

.c7 {
  width: 32px;
  height: 32px;
  margin-bottom: 8px;
}

.c1 {
  position: fixed;
  z-index: 3;
  cursor: pointer;
  width: 24px;
  height: 24px;
  -webkit-transition: all 0.2s cubic-bezier(0.34,1.56,0.64,1);
  transition: all 0.2s cubic-bezier(0.34,1.56,0.64,1);
  top: 16px;
  left: 16px;
  -webkit-transform: translate3d(-50px,0px,0px);
  -ms-transform: translate3d(-50px,0px,0px);
  transform: translate3d(-50px,0px,0px);
}

.c1 svg {
  pointer-events: none;
  fill: currentColor;
}

.c2 {
  position: fixed;
  z-index: 3;
  cursor: pointer;
  width: 24px;
  height: 24px;
  -webkit-transition: all 0.2s cubic-bezier(0.34,1.56,0.64,1);
  transition: all 0.2s cubic-bezier(0.34,1.56,0.64,1);
  top: 16px;
  left: 16px;
  -webkit-transform: scale(1);
  -ms-transform: scale(1);
  transform: scale(1);
  opacity: 1;
}

.c2 svg {
  pointer-events: none;
  fill: currentColor;
}

.c5 {
  -webkit-transition: -webkit-transform 0.2s cubic-bezier(0.165,0.84,0.44,1);
  -webkit-transition: transform 0.2s cubic-bezier(0.165,0.84,0.44,1);
  transition: transform 0.2s cubic-bezier(0.165,0.84,0.44,1);
  position: fixed;
  left: 0;
  top: 0;
  bottom: 0;
  width: 300px;
  z-index: 1;
  will-change: transform;
}

.c10 {
  padding: 32px 0;
  min-width: 0;
  position: absolute;
  background: #fff;
  right: 0;
  left: 0;
  height: 100vh;
  -webkit-transition: left 0.3s cubic-bezier(0.165,0.84,0.44,1);
  transition: left 0.3s cubic-bezier(0.165,0.84,0.44,1);
  will-change: left;
}

.c0 {
  background: #fff;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c0 .c4 {
  -webkit-transform: translateX(0);
  -ms-transform: translateX(0);
  transform: translateX(0);
}

@media screen and (max-width:376px) {
  .c2 {
    right: 16px;
    left: initial;
  }
}

@media (min-width:768px) {
  .c10 {
    padding: 0 32px;
    left: 300px;
  }
}

@media screen and (max-width:376px) {
  .c0 .c4 {
    right: 0;
    width: auto;
  }
}

<div
  className="c0"
>
  <div
    className="c1"
    onClick={[Function]}
  >
    <svg
      id="open"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M2 6a1 1 0 011-1h18a1 1 0 011 1 1 1 0 01-1 1H3a1 1 0 01-1-1zm0 6a1 1 0 011-1h18a1 1 0 011 1 1 1 0 01-1 1H3a1 1 0 01-1-1zm1 5a1 1 0 00-1 1 1 1 0 001 1h18a1 1 0 001-1 1 1 0 00-1-1H3z"
      />
    </svg>
  </div>
  <div
    className="c2"
    onClick={[Function]}
  >
    <svg
      id="close"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M6.23 4.81A1 1 0 004.8 6.23L10.6 12 4.8 17.77a1 1 0 101.42 1.42L12 13.4l5.78 5.78a1 1 0 001.4-1.42L13.42 12l5.78-5.77a1 1 0 10-1.41-1.42L12 10.6 6.23 4.8z"
      />
    </svg>
  </div>
  <aside
    className="c3 c4 c5"
  >
    <div
      className="c6"
    >
      <div
        className="c7"
      >
        <svg
          viewBox="0 0 44 44"
          xmlns="http://www.w3.org/2000/svg"
        >
          <rect
            fill="#7000E3"
            height="44"
            rx="3.9"
            width="44"
          />
          <path
            d="M28 11.3c-2.3 0-4.6 1-6 3-1.4-2-3.7-3-6-3-4.3 0-7.8 3.3-7.8 7.6 0 6.5 9.3 15 13.8 15s13.8-8.5 13.8-15c0-4.4-3.5-7.7-7.8-7.7"
            fill="#FF671B"
          />
        </svg>
      </div>
      <div
        className="c8"
      />
    </div>
    <div
      className="c9"
    />
  </aside>
  <main
    className="c10"
  />
</div>
`;
